src/middlewares/destTransformCompactedPayloadV1.ts

```diff
@@ -0,0 +1,23 @@
+import { Context, Next } from 'koa';
+import { ProcessorCompactedTransformationRequest, ProcessorTransformationRequest } from '../types';
+
+/**
+ * Middleware to transform a compacted payload for destination transformations
+ * to its decompacted form, as it is expected by the destination transformers.
+ * Compacted payloads are used to reduce the size of the payload by removing
+ * connection and destination details from each input and instead providing them
+ * in separate lookup tables.
+ */
+export class DestTransformCompactedPayloadV1Middleware {
+  public static async handle(ctx: Context, next: Next): Promise<void> {
+    if (ctx.get('x-content-format') === 'json+compactedv1') {
```

Comment:
Are we planning to add metrics on usage of this structure in rudder-server side?